optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/impl/result/SolverBenchmarkResult.java

```diff
@@ -257,10 +266,11 @@ public SingleBenchmarkResult findSingleBenchmark(ProblemBenchmarkResult problemB
     }
 
     public String getSolverConfigAsHtmlEscapedXml() {
-        // TODO reuse a single XStream instance for the entire report
-        XStream xStream = XStreamConfigReader.buildXStream();
-        xStream.setMode(XStream.NO_REFERENCES);
```

Comment:
(thinking aloud, feel free to ignore)
This was there to avoid xstream ID references in the html report that shows the source XML.
Basically, the source XML must be copyable and not introduce any changes from having been parsed and unparsed, except for inheriting the inheritedSolver of the benchmark xml.
So no id's add that weren't there, no null fields becoming defaulted fields etc.

I presume this is all fine, as IIRC we do have a test that covers it.

Reply:
Thanks for pointing this out; with JAXB one cannot turn IDs on and off, and there is one place where we have to use the reference, as otherwise, aggregation won't work.

Do you think it blocks merging or can this be resolved in a separate JIRA?

Reply:
https://issues.redhat.com/browse/PLANNER-2063